app/models/user.rb

```diff
@@ -0,0 +1,35 @@
+require 'data_mapper'
+require 'dm-postgres-adapter'
+require 'bcrypt'
+
+class User
+
+  include DataMapper::Resource
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_confirmation_of :password
+  validates_presence_of :password, :username, :name, :email
+  validates_uniqueness_of :email, :username
+
+  property :id, Serial
+  property :email, String
```

Comment:
You need to specify `unique: true` here instead of relying on the `validates_uniqueness_of :email`